lib/web/fetch/decompress.js

```diff
@@ -0,0 +1,95 @@
+const { pipeline } = require('node:stream')
+const zlib = require('node:zlib')
+const { redirectStatusSet, nullBodyStatus } = require('./constants')
+const { createInflate } = require('./util')
+
+/**
+ * Decompress the given response body.
+ * @param {Request} request
+ * @param {Response} response
+ * @returns {ReadableStream<Uint8Array> | null}
+ */
+function decompress (request, response) {
```

Comment:
I think decompressStream is sufficient

Reply:
I'd like to have a designated utility that performs the decompression _including_ the handling of redirect responses, etc. `decompressStream` implies you handle that yourself.

Reply:
It looks to me like you are targeting the Fetch api response.
In this case, have you already been redirected?

Reply:
@tsctx, I'm moving the existing content encoding logic to this utility function. In the existing logic, Undici does accept the response, which is the redirect response, not the redirected response. This is correct. This utility must also decide if decompression is unnecessary if the response is a redirect response.